save-backend/src/main/kotlin/com/saveourtool/save/backend/repository/ExecutionRepository.kt

```diff
@@ -45,8 +48,10 @@ interface ExecutionRepository : BaseEntityRepository<Execution> {
     fun findTopByProjectNameAndProjectOrganizationNameOrderByStartTimeDesc(name: String, organizationName: String): Optional<Execution>
 
     /**
-     * @param project to find execution
+     * @param id if the execution
      * @return execution
      */
-    fun findTopByProjectOrderByStartTimeDesc(project: Project): Execution?
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @QueryHints(QueryHint(name = "javax.persistence.lock.timeout", value = "10000"))
```

Comment:
Can we make it configurable via properties? In case the DB is too slow

Reply:
looks like we can't -- it's `javax.persistence.QueryHint` and it can't take a Spring's parameter as input